### PR TITLE
Lunch menu UI test with Stubs

### PR DESCRIPTION
**Pull request by dbacinski**

### CODE REVIEW

File: app/src/androidTest/groovy/com/macoscope/ketchuplunch/RecyclerViewMatcher.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        this.recyclerViewId = recyclerViewId;
+    }
+
+    public Matcher<View> atPosition(final int position) {
+        return atPositionOnView(position, -1);
+    }
+
+    public Matcher<View> atPositionOnView(final int position, final int targetViewId) {
+
+        return new TypeSafeMatcher<View>() {
```
Comment (pziomacki) [inline app/src/androidTest/groovy/com/macoscope/ketchuplunch/RecyclerViewMatcher.java]: Create new class that extends TypeSafeMatcher

**Comment by pziomacki** [comment]
-1

File: app/src/androidTest/groovy/com/macoscope/ketchuplunch/graybox/LunchMenuWithStubsSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ScriptClient scriptClientMock = Mockito.mock(ScriptClient)
+
+    def setup() {
+        LunchModule.scriptClient = scriptClientMock
+    }
+
+    def "displays menu for Monday"() {
+        given:
+            ArrayList<Map<String, Object>> apiResult = new ArrayList<Map<String, Object>>()
+            apiResult.add(["name": "Onion Soup", "count": 1 as BigDecimal, "totalCount": 3 as BigDecimal])
```
Comment (Tomekkk) [inline app/src/androidTest/groovy/com/macoscope/ketchuplunch/graybox/LunchMenuWithStubsSpec.groovy]: I don't know if this position is needed if not check in then block.

Comment (dbacinski) [inline app/src/androidTest/groovy/com/macoscope/ketchuplunch/graybox/LunchMenuWithStubsSpec.groovy]: checking 2 elements is redundant, I have to display at least 2 to make use of RecyclerViewMatcher, will change name of first element

File: app/src/androidTest/groovy/com/macoscope/ketchuplunch/RecyclerViewMatcher.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    static class ItemViewMatcher extends TypeSafeMatcher<View> {
+
+        public static final int ITEM_VIEW_ID = -1;
+
+        private final int position;
+        private final int targetViewId;
+        private int recyclerViewId;
+        Resources resources;
+        View itemView;
```
Comment (pziomacki) [inline app/src/androidTest/groovy/com/macoscope/ketchuplunch/RecyclerViewMatcher.java]: Add private modifiers for resources and itemView

**Comment by pziomacki** [comment]
+2

**Pull request merged**